build-tools/xa-prep-tasks/external-git-dependencies.targets

```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <UsingTask AssemblyFile="..\..\bin\Build$(Configuration)\xa-prep-tasks.dll" TaskName="Xamarin.Android.BuildTools.PrepTasks.Git" />
+  <UsingTask AssemblyFile="..\..\bin\Build$(Configuration)\xa-prep-tasks.dll" TaskName="Xamarin.Android.BuildTools.PrepTasks.ParseExternalGitDependencies" />
+  <UsingTask AssemblyFile="..\..\bin\Build$(Configuration)\xa-prep-tasks.dll" TaskName="Xamarin.Android.BuildTools.PrepTasks.CheckoutExternalGitDependency" />
+  <PropertyGroup>
+    <ExternalSourceDependencyDirectory>$(XamarinAndroidSourcePath)\external</ExternalSourceDependencyDirectory>
+  </PropertyGroup>
+  <Target Name="ParseExternalFile">
+    <ParseExternalGitDependencies 
+        ExternalFilePath="$(XamarinAndroidSourcePath)\.external" >
+      <Output TaskParameter="ExternalGitDependencies" ItemName="Externals" />
+    </ParseExternalGitDependencies>
+  </Target>
+  <Target Name="CheckoutExternalGitSources"
+      DependsOnTargets="ParseExternalFile">
+    <CheckoutExternalGitDependency
+        Condition="'%(Externals.Name)'=='monodroid'"
```

Comment:
Why this `Condition` value?  Wouldn't we *want* to check out everything within `.external`?

Reply:
My thinking was that we may eventually want to process certain items declared in `.external` in different ways. Perhaps we would eventually want to be able to use a branch/hash name from an entry in this file to calculate a URI to download a bundle or artifact from, rather than cloning it's source?

Maybe the better approach would be to add this condition in the future if/when such a use case arises.

Reply:
Yes, please add the condition in the future when such a use case arises.